### PR TITLE
Ensure underscore can be run in a Node VM

### PR DESCRIPTION
**Pull request by megawac**
Discovered during https://github.com/caolan/async/pull/804

### CODE REVIEW

**Comment by akre54** [comment]
I'm not sure I understand. Wouldn't you be using CommonJS (and thus `module.exports` instead of `root`) in a node environment? Why would you be using `_.noConflict()` at all?

File: test/utility.js [hunk 1]
```diff
@@ -14,6 +14,34 @@
 
   });
 
+  test('noConflict', function() {
```
Comment (akre54) [inline test/utility.js]: :+1:  this test

**Comment by megawac** [comment]
> Wouldn't you be using CommonJS (and thus module.exports instead of root) in a node environment? 
No, this is for the node [vm module](https://nodejs.org/api/vm.html) (sandboxes)

> Why would you use _.noConflict() at all?

You probably would't; the bigger issue is we're not correctly detecting the global, and thus the export for the vm, `root._ = _;` doesn't actually work.

File: test/utility.js [hunk 2]
```diff
@@ -14,6 +14,34 @@
...
+    var underscore = _.noConflict();
+    equal(this._, void 0);
```
Comment (michaelficarra) [inline test/utility.js]: To avoid strict mode problems, test `typeof _` instead.

**Pull request merged**